staker/multi_protocol_staker.go

```diff
@@ -0,0 +1,207 @@
+package staker
+
+import (
+	"context"
+	"time"
+
+	"github.com/OffchainLabs/bold/solgen/go/bridgegen"
+	boldrollup "github.com/OffchainLabs/bold/solgen/go/rollupgen"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/staker/txbuilder"
+	"github.com/offchainlabs/nitro/util/headerreader"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+var assertionCreatedId common.Hash
+
+func init() {
+	rollupAbi, err := boldrollup.RollupCoreMetaData.GetAbi()
+	if err != nil {
+		panic(err)
+	}
+	assertionCreatedEvent, ok := rollupAbi.Events["AssertionCreated"]
+	if !ok {
+		panic("RollupCore ABI missing AssertionCreated event")
+	}
+	assertionCreatedId = assertionCreatedEvent.ID
+}
+
+type MultiProtocolStaker struct {
+	stopwaiter.StopWaiter
+	bridge     *bridgegen.IBridge
+	oldStaker  *Staker
+	boldStaker *BOLDStaker
+}
+
+func NewMultiProtocolStaker(
+	l1Reader *headerreader.HeaderReader,
+	wallet ValidatorWalletInterface,
+	callOpts bind.CallOpts,
+	config L1ValidatorConfigFetcher,
+	blockValidator *BlockValidator,
+	statelessBlockValidator *StatelessBlockValidator,
+	stakedNotifiers []LatestStakedNotifier,
+	confirmedNotifiers []LatestConfirmedNotifier,
+	validatorUtilsAddress common.Address,
+	bridgeAddress common.Address,
+	fatalErr chan<- error,
+) (*MultiProtocolStaker, error) {
+	oldStaker, err := NewStaker(
+		l1Reader,
+		wallet,
+		callOpts,
+		config,
+		blockValidator,
+		statelessBlockValidator,
+		stakedNotifiers,
+		confirmedNotifiers,
+		validatorUtilsAddress,
+		fatalErr,
+	)
+	if err != nil {
+		return nil, err
+	}
+	bridge, err := bridgegen.NewIBridge(bridgeAddress, oldStaker.client)
+	if err != nil {
+		return nil, err
+	}
+	return &MultiProtocolStaker{
+		oldStaker:  oldStaker,
+		boldStaker: nil,
+		bridge:     bridge,
+	}, nil
+}
+
+func (m *MultiProtocolStaker) IsWhitelisted(ctx context.Context) (bool, error) {
+	return m.oldStaker.IsWhitelisted(ctx)
```

Comment:
This isn't accurate if bold is enabled.
It seems like the only reason this function is public is to print a log-line in arbnode's createNodeImpl.
Making this function accurate is a little compicated and not worth it.
Turn the function into private for the old staker, and remove "whitelisted" from the logline in arbnode.

Reply:
DOne.